bot/cogs/allow_deny_lists.py

```diff
@@ -0,0 +1,144 @@
+import logging
+
+from discord import Colour, Embed
+from discord.ext.commands import BadArgument, Cog, Context, group
+
+from bot import constants
+from bot.api import ResponseCodeError
+from bot.bot import Bot
+from bot.converters import ValidAllowDenyListType
+from bot.pagination import LinePaginator
+from bot.utils.checks import with_role_check
+
+log = logging.getLogger(__name__)
+
+
+class AllowDenyLists(Cog):
+    """Commands for blacklisting and whitelisting things."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    async def _add_data(self, ctx: Context, allowed: bool, list_type: ValidAllowDenyListType, content: str) -> None:
+        """Add an item to an allow or denylist."""
+        payload = {
+            'allowed': allowed,
+            'type': list_type,
+            'content': content,
+        }
+        allow_type = "whitelist" if allowed else "blacklist"
+
+        # Try to add the item to the database
+        try:
+            item = await self.bot.api_client.post(
+                "bot/allow_deny_lists",
+                json=payload
+            )
+        except ResponseCodeError as e:
+            if e.status == 500:
```

Comment:
So would I, but that's not what it does. The constraint violations produce 500s. This might be solvable by making some change on the Django side, but I couldn't figure out what that change would have to be.

Reply:
Yeah, this needs to be figured out, I think. It doesn't sound normal. If you get 500s it means DRF isn't handling the constraint for some reason.

Reply:
Okay, looks like this is happening because I am setting this constraint in a Django vanilla way, not in a DRF way. 

Fixed on the site in https://github.com/python-discord/site/pull/371/commits/0d7c95a80692f15c36b36152034e3bbe54e3c37c

Reply:
Fixed on the bot in 1b6be86